homeassistant/components/mqtt/__init__.py

```diff
@@ -186,13 +200,17 @@ def embedded_broker_deprecated(value):
                     vol.Optional(CONF_PROTOCOL, default=DEFAULT_PROTOCOL): vol.All(
                         cv.string, vol.In([PROTOCOL_31, PROTOCOL_311])
                     ),
-                    vol.Optional(CONF_WILL_MESSAGE): MQTT_WILL_BIRTH_SCHEMA,
-                    vol.Optional(CONF_BIRTH_MESSAGE): MQTT_WILL_BIRTH_SCHEMA,
+                    vol.Optional(
+                        CONF_WILL_MESSAGE, default=dict(DEFAULT_WILL)
```

Comment:
Why copy it?

Reply:
I want to avoid hard to find bugs because the `DEFAULT_WILL` ends up modified somewhere.

Reply:
Voluptuous will already copy the dict. 

![image](https://user-images.githubusercontent.com/1444314/86483780-9f1c2a80-bd09-11ea-80db-6913462f4692.png)

Reply:
Hmm, but:
![image](https://user-images.githubusercontent.com/14281572/86493816-41242d00-bd73-11ea-8021-efe63470953a.png)

Reply:
You're validating against `dict` in your test, not `{}`. When you do `{}` it will check the keys and their values and make a copy.

Reply:
You're right, I dropped the useless copy.